onebusaway-android/src/main/java/org/onebusaway/android/ui/arrivals/components/EtaStripLoadMore.kt

```
@@ -80,3 +87,25 @@ internal fun Modifier.acknowledgeVersion(version: State<Long>, into: MutableLong
+ * than over/undershot. [target] is read inside `snapshotFlow`, so it must be a plain synchronous read
+ * of snapshot state, not a suspend function. Never contests a real user drag: losing the scrollable
+ * mutex to one just waits for the strip to go idle before re-deriving the target on the next iteration
+ * (the caller's own nested-scroll hook is what ends the chase for good, by cancelling this coroutine).
+ */
+internal suspend fun ScrollState.glideTo(target: () -> Int?) {
+    while (true) {
+        snapshotFlow { target() }.filterNotNull().first { it != value }
+        val next = target() ?: continue
+        try {
+            animateScrollTo(next)
+        } catch (cause: CancellationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)